kernel/rtc/src/lib.rs

```diff
@@ -185,10 +190,10 @@ fn log2(value: usize) -> usize {
 
 /// sets the period of the RTC interrupt. 
 /// `rate` must be a power of 2, between 2 and 8192 inclusive.
-pub fn set_rtc_frequency(rate: usize) -> Result<(), ()> {
+pub fn set_rtc_frequency(rate: usize) -> Result<(), RtcError> {
```

Comment:
Let's use a more specific error type (empty enum or struct) that is strictly about an invalid frequency rate value. `InvalidInput` is vague.